KSE-1786 | Migration: Jenkins -> Semaphore

pbadani opened the pull request:
### Description 
Jenkins migration to semaphore.
This is the same as the autogenerated PR https://github.com/confluentinc/ksql/pull/10102/files
But targeted to 4.1.x branch. This will be pint merged.

### Testing done 
_Describe the testing strategy. Unit and integration tests are expected for any behavior changes._

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")
- [ ] Do these changes have compatibility implications for rollback? If so, ensure that the ksql [command version](https://github.com/confluentinc/ksql/blob/master/ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/Command.java#L41) is bumped.

pbadani committed:
KSE-1786 | Migration: Jenkins -> Semaphore

cprovencher commented:
Please add "mvnSkipDeploy = true" in the Jenkinsfile also. Like here https://github.com/confluentinc/ksql/pull/10102/files#diff-e6ffa5dc854b843b3ee3c3c28f8eae2f436c2df2b1ca299cca1fa5982e390cf8R27

sriman-confluent commented on .semaphore/semaphore.yml:
```diff
@@ -0,0 +1,89 @@
+  prologue:
+    commands:
+      - checkout
+      - . set-cp-java-version
```
FYI, this means that Semaphore will run builds/tests using Java17 for branches starting at 4.1.x. If you need to use Java8 or Java11 for builds on older branches, you can use this PR as reference for how to alter the java version: https://github.com/confluentinc/confluent-security-plugins/pull/647/files

pbadani committed:
KSE-1786 | Set mvnSkipDeploy = true

pbadani committed:
KSE-1786 | Use java 8 for semaphore builds.

pbadani commented:
Addressed all the review comments. 
Closing this in favour of https://github.com/confluentinc/ksql/pull/10173
To be merged from branch 6.2.x onwards.
